src/lib/components/WalletConnect.svelte

```diff
@@ -0,0 +1,48 @@
+<script lang="ts">
+	import { connectWallet, disconnectWallet, walletAddress, network } from '$lib/viem';
+
+	async function handleConnect() {
+		try {
+			await connectWallet();
+		} catch (error) {
+			console.error('Failed to connect wallet:', error);
+			alert('Failed to connect wallet. Make sure MetaMask is installed and try again.');
```

Comment:
Nitpick, `MetaMask` isn't the only option here. Maybe say "Make sure a browser wallet is installed"